File: internal/commit/commit.go
package commit

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"os/exec"
	"strings"

	"github.com/Evan-acg/GitConventionalCommits/internal/color"
)

type Entry struct {
	Type    string   `json:"type"`
	Scope   string   `json:"scope"`
	Message string   `json:"message"`
	Files   []string `json:"files,omitempty"`
	Detail  string   `json:"detail,omitempty"`
}

type AIResponse struct {
	Reason string  `json:"reason"`
	Data   []Entry `json:"data"`
}

func ConfirmEntry(entry Entry) bool {
	msg := FormatMessage(entry)
	fmt.Println(color.CyanS("生成的提交消息:"))
	fmt.Println(color.BoldS(msg))
	if len(entry.Files) > 0 {
		fmt.Println(color.CyanS("\n关联文件:"))
		for _, f := range entry.Files {
			fmt.Println("  " + f)
		}
	}
	fmt.Print(color.CyanS("\n确认提交？回复 ok 执行: "))

	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	return strings.TrimSpace(scanner.Text()) == "ok"
}

func FormatMessage(entry Entry) string {
	msg := entry.Type + "(" + entry.Scope + "): " + entry.Message
	if entry.Detail != "" {
		msg += "\n\n" + entry.Detail
	}
	return msg
}

func Stage() error {
	return exec.Command("git", "add", "-A").Run()
}

func StageFiles(files []string) error {
	args := append([]string{"add"}, files...)
	return exec.Command("git", args...).Run()
}

func UnstageFiles(files []string) error {
	args := append([]string{"restore", "--staged"}, files...)
	return exec.Command("git", args...).Run()
}

func Commit(msg string) error {
	parts := strings.SplitN(msg, "\n\n", 2)
	args := []string{"commit", "-m", parts[0]}
	if len(parts) > 1 && parts[1] != "" {
		args = append(args, "-m", parts[1])
	}
	cmd := exec.Command("git", args...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	return cmd.Run()
}

func CommitFiles(msg string, files []string) error {
	parts := strings.SplitN(msg, "\n\n", 2)
	args := []string{"commit", "-m", parts[0]}
	if len(parts) > 1 && parts[1] != "" {
		args = append(args, "-m", parts[1])
	}
	args = append(args, "--")
	args = append(args, files...)
	cmd := exec.Command("git", args...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	return cmd.Run()
}

func ParseEntries(raw string) ([]Entry, string) {
	raw = strings.TrimSpace(raw)
	if raw == "" {
		return nil, ""
	}

	var resp AIResponse
	if err := json.Unmarshal([]byte(raw), &resp); err != nil {
		return nil, ""
	}

	return resp.Data, resp.Reason
}

File: src/commit.rs
use serde::{Deserialize, Serialize};
use std::process::Command;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Entry {
    #[serde(rename = "type")]
    pub entry_type: String,
    pub scope: String,
    pub message: String,
    #[serde(default)]
    pub files: Vec<String>,
    #[serde(default)]
    pub detail: String,
}

#[derive(Debug, Deserialize)]
pub struct AiResponse {
    pub reason: String,
    pub data: Vec<Entry>,
}

pub fn format_message(entry: &Entry) -> String {
    let msg = format!("{}({}): {}", entry.entry_type, entry.scope, entry.message);
    if entry.detail.is_empty() {
        msg
    } else {
        format!("{msg}\n\n{}", entry.detail)
    }
}

pub fn confirm_entry(entry: &Entry) -> bool {
    let msg = format_message(entry);
    println!("\n生成的提交消息:");
    println!("{msg}");
    if !entry.files.is_empty() {
        println!("\n关联文件:");
        for f in &entry.files {
            println!("  {f}");
        }
    }
    print!("\n确认提交？回复 ok 执行: ");
    std::io::Write::flush(&mut std::io::stdout()).ok();
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).ok();
    input.trim() == "ok"
}

pub fn stage_all() -> anyhow::Result<()> {
    let status = Command::new("git").args(["add", "-A"]).status()?;
    if !status.success() {
        anyhow::bail!("git add -A 失败");
    }
    Ok(())
}

pub fn stage_files(files: &[String]) -> anyhow::Result<()> {
    let mut args = vec!["add".to_string()];
    args.extend_from_slice(files);
    let status = Command::new("git").args(&args).status()?;
    if !status.success() {
        anyhow::bail!("git add 失败");
    }
    Ok(())
}

pub fn commit(msg: &str) -> anyhow::Result<()> {
    let parts: Vec<&str> = msg.splitn(2, "\n\n").collect();
    let mut args = vec!["commit", "-m", parts[0]];
    if parts.len() > 1 && !parts[1].is_empty() {
        args.push("-m");
        args.push(parts[1]);
    }
    let status = Command::new("git").args(&args).status()?;
    if !status.success() {
        anyhow::bail!("git commit 失败");
    }
    Ok(())
}

pub fn parse_entries(raw: &str) -> (Vec<Entry>, String) {
    let raw = raw.trim();
    if raw.is_empty() {
        return (vec![], String::new());
    }
    match serde_json::from_str::<AiResponse>(raw) {
        Ok(resp) => (resp.data, resp.reason),
        Err(_) => (vec![], String::new()),
    }
}
